src/db.rs

```diff
@@ -1854,12 +1858,47 @@ mod tests {
 				expected_count -= 1;
 			}
 		}
-		assert_eq!(col_stats.total_values, expected_count);
+		if ref_counted {
+			let mut state: BTreeMap<Vec<u8>, Option<(Vec<u8>, usize)>> = Default::default();
```

Comment:
Genuine question: What is the difference in `states` between a `None` value and the value not being in the `BTreeMap`? I would believe that `BTreeMap<Vec<u8>, (Vec<u8>, usize)>` would be as good but I'm maybe missing something.

Reply:
I ask me the same thing yesterday (actually did start changing it). Having None is needed to assert the value a few line bellow do not exists in the db (so if a removal fail we would see the value still in db).